Nimble/Matchers/AllPass.swift

```diff
@@ -0,0 +1,99 @@
+import Foundation
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (passFunc: (T?) -> Bool) -> NonNilMatcherFunc<U> {
+        return allPass("pass a condition", passFunc)
+}
+
+public func allPass<T,U where U: SequenceType, U.Generator.Element == T>
+    (passName:String, passFunc: (T?) -> Bool) -> NonNilMatcherFunc<U> {
+        return createAllPassMatcher() {
+            expression, failureMessage in
+            failureMessage.postfixMessage = passName
+            return passFunc(expression.evaluate())
+        }
+}
+
+public func allPass<U,V where U: SequenceType, V: NonNilBasicMatcher, U.Generator.Element == V.ValueType>
+    (matcher: V) -> NonNilMatcherFunc<U> {
+        let wrapper = NonNilMatcherWrapper(NonNilBasicMatcherWrapper(matcher))
+        return createAllPassMatcher() {wrapper.matches($0, failureMessage: $1)}
+}
+
+public func allPass<U,V where U: SequenceType, V: BasicMatcher, U.Generator.Element == V.ValueType>
+    (matcher: V) -> NonNilMatcherFunc<U> {
+        let wrapper = BasicMatcherWrapper(matcher: matcher)
+        return createAllPassMatcher() {wrapper.matches($0, failureMessage: $1)}
+}
+
+public func allPass<U,V where U: SequenceType, V: Matcher, U.Generator.Element == V.ValueType>
```

Comment:
Yeah, there needs to be some changes to simplify the internals. I think the removal a BasicMatcher and NonNilBasicMatcher would be preferred in some way. There also needs a reinvestigation if the swift compiler still gives an obscure message when a generic does match optionals. having the matcher be able to specify that while Nimble's `Expectation` and matcher protocols be ignorant of that detail would be best.

All that stuff are just floating in my head right now. For various personal reasons, I don't have the available free time yet.